test/puppetlabs/puppetdb/testutils/jetty.clj

```diff
@@ -61,42 +62,55 @@
     (spit logback-file (log-config log-file "ERROR"))
     [log-file (assoc-in config [:global :logging-config] logback-file)]))
 
-(defn current-port
-  "Given a trapperkeeper server, return the port of the running jetty instance.
-  Note there can be more than one port (i.e. SSL + non-SSL connector). This only
-  returns the first one."
-  [server]
-  (-> @(tka/app-context server)
-      (get-in [:WebserverService :jetty9-servers :default :server])
-      .getConnectors
-      first
-      .getLocalPort))
+(defn open-port-num
+  "Returns a currently open port number"
+  []
+  (with-open [s (java.net.ServerSocket. 0)]
+    (.getLocalPort s)))
+
+(defn assoc-open-port
```

Comment:
I don't think so. The convention I follow is functions that directly modify mutable variables have a !, like swap! or reset! in clojure core. Things that have side affects like DB calls or I/O, I don't use that (i.e. spit in clojure core)